scipy/stats/tests/test_multivariate.py

```diff
@@ -1870,6 +1870,14 @@ def test_allow_singular(self):
         args = dict(loc=[0,0], shape=[[0,0],[0,1]], df=1, allow_singular=False)
         assert_raises(np.linalg.LinAlgError, multivariate_t, **args)
 
+    @pytest.mark.parametrize("size", [(10, 3), (5, 6, 4, 3)])
```

Comment:
Maybe should test edge cases where `size` is `None`, `1`, or `(1,)`? I imagine `(1,)` would be incorrect because of the squeeze? You don't necessarily have to fix that here if it's a problem for most multivariate distributions if you think it needs a separate issue.

Reply:
> Maybe should test edge cases where `size` is `None`, `1`, or `(1,)`?

If we are comparing to NumPy RNG, it would give incorrect shape for all these cases. For `None`, incorrect output is given for one-dimensional distribution (NumPy gives `(1, )` output while `multivariate_t` gives a scalar output). For `1` and `(1, )`, `multivaraite_t` squeezes those dimensions that contain a `1` while NumPy preserves it.

> I imagine `(1,)` would be incorrect because of the squeeze?

Yes. `multivariate_t` will squeeze any dimension with `1` in it. So, for example, `(1,1,1,1,1)` and `(1,2,1,3,1,4)` will give output of shape `()` and `(2,3,4)` respectively. This is because `_squeeze_output` squeezes all the dimensions with `1` in them.


> You don't necessarily have to fix that here if it's a problem for most multivariate distributions if you think it needs a separate issue.

I think it needs a separate issue because it is problem for many other distributions (`multivariate_normal`, `wishart`, `invwishart`, etc) and multiple methods (`rvs`, `pdf`, etc) might be broken.

> To make edge cases easier to test, perhaps use the NumPy RNG as a reference for the shape.

That's a nice idea. But NumPy doesn't have a sampler for `multivariate_t` distribution. But the shapes against which I check the output follow NumPy's rules for samples from the multivariate distributions.